### PR TITLE
Speedy Snake Game

### PR DESCRIPTION
**Pull request by ItsAStarryKnight**
# Your checklist for this pull request

## About your game

What is your game about? 
> This is a new take on the classic snake game. Instead of growing when it eats, this speedy snake increases in speed after eating apples, creating a unique mechanic that is a novel way to play a classic game!

How to play your game?
> Press WASD to move, L to restart, J to enable hard mode (speed increases more for each apple eaten) and K to enable easy mode

Coding experience:
> I have been messing around with programming in different languages for a couple of years, learning python and java in some of my high school classes. I also have learned some javascript/html/css to build websites, although I am still learning in that department :)

## Code
Check off the items that are true.
- [x] The game was made using the [Sprig editor](https://editor.sprig.hackclub.com/).
- [x] The game is placed in the in the [`/games` directory](https://github.com/hackclub/sprig/tree/main/games).
- [x] The code is significantly different from all other games in the [Sprig gallery](https://sprig.hackclub.com/gallery) (except for games labeled "demo").
- [x] The game runs without errors. 
- [x] The name of the file/game contains only alphanumeric characters, `-`s, or `_`s.
- [x] The game name is not the same as the others from [gallery](https://sprig.hackclub.com/gallery)

## Image (If an image is used)

- [x] The image is in the [`/games/img` directory](https://github.com/hackclub/sprig/tree/main/games/img).
- [x] The name of the image matches the name of your file. Example: `sokoban_plus.js` -> `sokoban_plus.png`.

> Thanks for PR!

### CODE REVIEW

**Comment by LucasHT22** [comment]
Game very good! could you add more details in the sprite? snake eyes...

**Comment by ItsAStarryKnight** [comment]
I just updated, made new functionality for pause/play as well as making my sprites look more lively (both snake and wall!)

File: games/speedy_snake.js [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+}
+
+function checkState(){
+  var arr = getTile(getFirst(player).x,getFirst(player).y)
+  if(arr.length>1){
+    if(arr[1].type=="a"){
+      resetApple();
+    }
+  } else{
+    //console.log(getFirst(player).y, getFirst(player).x)
```
Comment (LucasHT22) [inline games/speedy_snake.js:219]: what is the functionality of this commented code?

Comment (ItsAStarryKnight) [reply games/speedy_snake.js:219]: The console.log was just for debugging, I can remove it now that it works fine!

**Comment by ItsAStarryKnight** [comment]
Ok I removed it - thanks for pointing it out!

**Review by LucasHT22 — APPROVED** [review]
Looks good to me! Congrats:)
If you're a teen, you can get your sprig here! https://sprig-order.hackclub.dev/


out of curiosity, how did you meet sprig?

**Comment by ItsAStarryKnight** [comment]
My friend actually found the github page, and he told me about it - I thought it was cool so tried it out!! So glad I did!

**Comment by LucasHT22** [comment]
nice!

**Pull request merged**